src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -26,33 +27,50 @@ object Scope {
     def getChildrenList(ast: Ast): Seq[Ast] = Option(ast.getAttribute("body")) collect {
       case list: AstList => list.asScala.toVector
     } getOrElse Seq.empty collect {
-      case ast: Ast => ast
+      case node: Ast
+        if node.getName == AstNodeName.Call ||
+          node.getName == AstNodeName.Scatter =>
+        node
+    }
+
+    /**
+     * Sets the child scopes on this parent from the ast.
+     * @param parentAst Parent ast.
+     * @param parentScope Parent scope.
+     */
+    def setChildScopes(parentAst: Ast, parentScope: Scope): Unit = {
+      //Increment index if needed
+      scopeIndexes.put(parentScope.getClass, scopeIndexes.getOrElse(parentScope.getClass, 0) + 1)
```

Comment:
Does the the scope index need to be set on the parent in setChildScopes?  That doesn't really seem like the responsibility of this method, and perhaps more appropriate at the call site?